service/src/main/java/uk/gov/hmcts/reform/fpl/service/DraftCMOService.java

```diff
@@ -0,0 +1,92 @@
+package uk.gov.hmcts.reform.fpl.service;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.commons.lang3.ObjectUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.ccd.client.model.CaseDetails;
+import uk.gov.hmcts.reform.fpl.model.CaseData;
+import uk.gov.hmcts.reform.fpl.model.CaseManagementOrder;
+import uk.gov.hmcts.reform.fpl.model.HearingBooking;
+import uk.gov.hmcts.reform.fpl.model.common.Element;
+import uk.gov.hmcts.reform.fpl.model.common.dynamic.DynamicList;
+import uk.gov.hmcts.reform.fpl.model.common.dynamic.DynamicListElement;
+import uk.gov.hmcts.reform.fpl.utils.HearingDateHelper;
+
+import java.time.LocalDate;
+import java.time.format.FormatStyle;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+import static org.apache.commons.lang3.StringUtils.isEmpty;
+
+@Service
+public class DraftCMOService {
+
+    private final ObjectMapper mapper;
+    private final DateFormatterService dateFormatterService;
+
+    @Autowired
+    public DraftCMOService(DateFormatterService dateFormatterService, ObjectMapper mapper) {
+        this.mapper = mapper;
+        this.dateFormatterService = dateFormatterService;
+    }
+
+    public DynamicList getHearingDatesDynamic(CaseDetails caseDetails) {
+        Map<String, Object> caseDataMap = caseDetails.getData();
+        CaseData caseData = mapper.convertValue(caseDataMap, CaseData.class);
+
+        List<Element<HearingBooking>> hearingDetails = caseData.getHearingDetails();
+
+        DynamicList hearingDatesDynamic = buildDynamicListFromHearingDetails(hearingDetails);
+        CaseManagementOrder caseManagementOrder = caseData.getCaseManagementOrder();
+
+        prePopulateHearingDateSelection(hearingDetails, hearingDatesDynamic, caseManagementOrder);
+
+        return hearingDatesDynamic;
+    }
+
+    private void prePopulateHearingDateSelection(
+        List<Element<HearingBooking>> hearingDetails,
+        DynamicList hearingDatesDynamic,
+        CaseManagementOrder caseManagementOrder) {
+        if (caseManagementOrder != null) {
+            String hearingDateId = ObjectUtils.isEmpty(caseManagementOrder) ? ""
```

Comment:
```suggestion
            String hearingDateId = isEmpty(caseManagementOrder) ? ""
```

Reply:
if we know caseManagementOrder is not null, we can assume data has been entered right? So do we need the empty check here at all?

Reply:
'isEmpty' method needs char sequence as input. ```    public static boolean isEmpty(final CharSequence cs) {
        return cs == null || cs.length() == 0;
    }```

Reply:
@swalker125  is suggesting to just do a static import of `ObjectUtils.isEmpty` instead

Reply:
Thanks @lewisbirks @swalker125  just done this change.